### PR TITLE
Rework test_plan_serialization_bwc to do roundtrip

### PR DESCRIPTION
**Pull request by carlopi**
Before this PR:
- "Generate serialized plans file" with GEN_PLAN_STORAGE set generated test/api/serialized_plans/serialized_plans.binary
- "Generate serialized plans file" without GEN_PLAN_STORAGE did nothing
- "Test deserialized plans from file" tested against test/api/serialized_plans/serialized_plans.binary

Afterwards
- "Generate serialized plans file" with GEN_PLAN_STORAGE set generates test/api/serialized_plans/serialized_plans.binary AND test it's deserialization
- "Generate serialized plans file" without GEN_PLAN_STORAGE generates test/api/serialized_plans/serialized_plans.temp.binary AND test it's deserialization
- "Test deserialized plans from file" tests against test/api/serialized_plans/serialized_plans.binary

Basically always rountrip test in "Generate serialized plans file" (either on real or temp file) and the same as before in "Test deserialized plans from file".
Also test on generation is run as part of serialization suite.
Main logic should be the same that @bleskes originally committed with minor refactoring.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,6 +39,7 @@ autoconf/aclocal.m4
 autoconf/autom4te.cache
 /compile_commands.json
 /.wasm
+/test/api/serialized_plans/serialized_plans.temp.binary
```
[the remark this answers is not in the document]
Comment (Mytherin) [reply .gitignore:42]: Can we use the `TestCreatePath` C++ function instead? That will create the path in the test folder which is automatically cleaned up afterwards

[the remark this answers is not in the document]
Comment (carlopi) [reply .gitignore]: Thanks, adapted!

**Review by Mytherin — APPROVED** [review]
Thanks, LGTM

**Comment by carlopi** [comment]
Single CI failure is unrelated: https://github.com/duckdb/duckdb/actions/runs/5208703631/jobs/9399017888?pr=7862#step:6:988
